espresso-support/core/src/main/java/com/novoda/espresso/AccessibilityViewMatchers.java

```diff
@@ -0,0 +1,187 @@
+package com.novoda.espresso;
+
+import android.content.res.Resources;
+import android.os.Build;
+import android.support.annotation.RequiresApi;
+import android.support.annotation.StringRes;
+import android.view.View;
+import android.view.accessibility.AccessibilityNodeInfo;
+
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.TypeSafeMatcher;
+
+import static org.hamcrest.Matchers.is;
+
+public class AccessibilityViewMatchers {
+
+    @RequiresApi(Build.VERSION_CODES.LOLLIPOP)
+    public static Matcher<? super View> withUsageHintOnClick(@StringRes final int resourceId) {
+        return new TypeSafeMatcher<View>() {
+
+            private String resourceName = null;
+            private String expectedText = null;
+
+            @Override
+            protected boolean matchesSafely(View view) {
+                if (!view.isClickable()) {
+                    return false;
+                }
+
+                if (null == this.expectedText) {
+                    try {
+                        expectedText = view.getResources().getString(resourceId);
+                        resourceName = view.getResources().getResourceEntryName(resourceId);
+                    } catch (Resources.NotFoundException ignored) {
+                        // view could be from a context unaware of the resource id.
+                    }
+                }
+                if (null != expectedText) {
+                    AccessibilityNodeInfo accessibilityNodeInfo = view.createAccessibilityNodeInfo();
+                    AccessibilityNodeInfo.AccessibilityAction clickAction = findAction(accessibilityNodeInfo, AccessibilityNodeInfo.ACTION_CLICK);
+
+                    return expectedText.equals(clickAction.getLabel());
+                } else {
+                    return false;
+                }
+            }
+
+            @Override
+            public void describeTo(Description description) {
+                description.appendText("is clickable and has custom usage hint for ACTION_CLICK from resource id: ");
+                description.appendValue(resourceId);
+                if (null != this.resourceName) {
+                    description.appendText("[");
+                    description.appendText(resourceName);
+                    description.appendText("]");
+                }
+                if (null != this.expectedText) {
+                    description.appendText(" value: ");
+                    description.appendText(expectedText);
+                }
+            }
+        };
+    }
+
+    @RequiresApi(Build.VERSION_CODES.LOLLIPOP)
+    public static Matcher<? super View> withUsageHintOnClick(CharSequence text) {
+        return withUsageHintOnClick(is(text));
+    }
+
+    @RequiresApi(Build.VERSION_CODES.LOLLIPOP)
+    public static Matcher<? super View> withUsageHintOnClick(final Matcher<? extends CharSequence> charSequenceMatcher) {
```

Comment:
I might be wrong but it seems there's a lot of common logic in all these matchers that could be probably be extracted into a common superclass (yeah I know, composition would be better but not sure it would work here), including the longpress matchers